docsTech/README.md

```diff
@@ -0,0 +1,69 @@
+
+## Where is the code?
+
+In folder `packages` you will find one subfolder per exported package (`@mui/x-data-grid`, `@mui/x-data-grid-pro`, ...)
+
+Since `@mui/x-data-grid-pro` is an extension of `@mui/x-data-grid`, it imports all the common features from `@mui/x-data-grid` and the internal logic are imported from `@mui/x-data-grid/internals` which is defined in `x-data-grid/src/internals/index.ts`.
+
+In each package, the most used folders are:
+
+- `components` which contains all the React components. These files are responsible for displaying the grid and listening for user interactions.
+- `hooks` which contains all the logic of the DataGrid.
+  - `hooks/core` defines logics for developers: initialization of the grid, error handling, translations, ...
+  - `hooks/utils` defines logics for contributors: manipulating the state, managing events, ...
+  - `hooks/features` defines logics for end-users: selection, filtering, editing, ...
+- `DataGrid[Pro]/useDataGrid[Pro]Component.tsx` is the file in which all the hooks are added to the grid
+- `colDef` contains the default values for each [column type](https://mui.com/components/data-grid/columns/#column-types): rendering cell, filter operators, sorting method, ...
+- `models` define the typescript interfaces of the gird
+
+## How does it work?
+
+![Workflow Scheme](./img/overviewSchemDark.svg#gh-dark-mode-only)
+![Workflow Scheme](./img/overviewSchemLight.svg#gh-light-mode-only)
+
+### Accessing state
+
+The react components can access the state with either selectors or getter methods.
+Here is an example.
+
+```js
+const apiRef = useGridApiContext();
+
+const filterModel = useGridSelector(apiRef, gridFilterModelSelector);
+const row = apiRef.current.getRow(id);
+```
+
+### Updating state
+
+To update the state, the component can either call a setter or publish an event. Here are examples with updating cell editing by setter, and handling cell key down by dispatching `GridEvents.cellKeyDown` event.
+
+```js
+apiRef.current.setEditCellValue(newParams, event);
+apiRef.current.publishEvent(GridEvents.cellKeyDown, params, event as any);
+```
+
+Publishing an event allows triggering multiple features. For example, `GridEvents.cellKeyDown` is listened by the focus, edit, and navigation feature hooks. Each of them defines an event handler responsible of modifying the state.
+
+```js
+useGridApiEventHandler(apiRef, GridEvents.cellKeyDown, handleCellKeyDown);
+```
+
+The other interest of events is that developers can listen to them by subscribing to the event for example [the documentation](https://mui.com/components/data-grid/events/) or by providing a `on<Event>` prop such as `onEditCellPropsChange`. Notice that all the events do not have an associated prop `on<Event>`
+
+## Specific features
+
+Here you will find more precision about how some features are designed.
+
+- [filtering](./filtering)
+- [virtualization](./virtualization)
```

Comment:
If we use the word "feature" for user-facing stuff  (like you did in `defines logics for end-users: selection, filtering, editing, ...`, and I fully agree)

Then maybe we should move the `virtualization` to a `Core concepts` paragraph.
In this paragraph I will also add pages about the plugin system (execution order, state init, pre-processor, ...)

Reply:
I was looking to have same structure as hook folders in which `virtualization` is a feature, which makes sense because, that's something developers can have access to if (by disabling it of modifying buffer)

In opposite, execution order, state init, or pre-processor should be transparent to developers.